### DISCUSSION TITLE
Adding parallel processing

### DISCUSSION THREAD

**Discussion by joaomdmoura** [crewAIInc/crewAI#1140] [Feature Requests]
Let's expand the processes!

**Comment by slavakurilyak** [comment]
+1 for Async Execution of Agents, not just Tasks

While Tasks can be executed asynchronously, calling the same Agent more than once is highly beneficial

Imagine a team of three Researcher Agents, each performing work in parallel before Research Manager evaluates the response from each

**Comment by tanevanwifferen** [comment]
I've got this use case where I'd like to take [this](https://github.com/e2b-dev/awesome-ai-agents) list of github projects, then to find out the last commit date for each project and return me a list of all project sorted by last activity date.

Ideally the tasks are parallelized, where each agent only gets the url to scrape and returns the date the repo was last updated. This saves immense amounts of context, and also makes sure the LLM doesn't run out of context halfway the url list.

Would be nice to use an AI for this instead of writing any code myself

**Comment by testcoplanet** [comment]
@joaomdmoura how is the latest status on tool calling? almost every agentic framwork is supporting it, but crewai
